frame/bounties/src/tests.rs

```diff
@@ -0,0 +1,937 @@
+	traits::{BlakeTwo256, IdentityLookup, BadOrigin},
+};
+
+impl_outer_origin! {
+	pub enum Origin for Test where system = frame_system {}
+}
+
+mod bounties {
+	// Re-export needed for `impl_outer_event!`.
+	pub use crate::*;
+}
+
+// impl_outer_event! {
+// 	pub enum Event for Test {
+// 		system<T>,
+// 		pallet_balances<T>,
+// 		pallet_treasury<T>,
+// 		bounties<T>,
+// 	}
+// }
+
+
+#[derive(Clone, Eq, PartialEq)]
+pub struct Test;
+parameter_types! {
+	pub const BlockHashCount: u64 = 250;
+	pub const MaximumBlockWeight: Weight = 1024;
+	pub const MaximumBlockLength: u32 = 2 * 1024;
+	pub const AvailableBlockRatio: Perbill = Perbill::one();
+}
+impl frame_system::Trait for Test {
+	type BaseCallFilter = ();
+	type Origin = Origin;
+	type Index = u64;
+	type BlockNumber = u64;
+	type Call = ();
+	type Hash = H256;
+	type Hashing = BlakeTwo256;
+	type AccountId = u128; // u64 is not enough to hold bytes used to generate bounty account
+	type Lookup = IdentityLookup<Self::AccountId>;
+	type Header = Header;
+	// type Event = Event;
+	type Event = ();
+	type BlockHashCount = BlockHashCount;
+	type MaximumBlockWeight = MaximumBlockWeight;
+	type DbWeight = ();
+	type BlockExecutionWeight = ();
+	type ExtrinsicBaseWeight = ();
+	type MaximumExtrinsicWeight = MaximumBlockWeight;
+	type AvailableBlockRatio = AvailableBlockRatio;
+	type MaximumBlockLength = MaximumBlockLength;
+	type Version = ();
+	type PalletInfo = ();
+	type AccountData = pallet_balances::AccountData<u64>;
+	type OnNewAccount = ();
+	type OnKilledAccount = ();
+	type SystemWeightInfo = ();
+}
+parameter_types! {
+	pub const ExistentialDeposit: u64 = 1;
+}
+impl pallet_balances::Trait for Test {
+	type MaxLocks = ();
+	type Balance = u64;
+	// type Event = Event;
+	type Event = ();
+	type DustRemoval = ();
+	type ExistentialDeposit = ExistentialDeposit;
+	type AccountStore = System;
+	type WeightInfo = ();
+}
+thread_local! {
+	static TEN_TO_FOURTEEN: RefCell<Vec<u128>> = RefCell::new(vec![10,11,12,13,14]);
+}
+pub struct TenToFourteen;
+impl Contains<u128> for TenToFourteen {
+	fn sorted_members() -> Vec<u128> {
+		TEN_TO_FOURTEEN.with(|v| {
+			v.borrow().clone()
+		})
+	}
+	#[cfg(feature = "runtime-benchmarks")]
+	fn add(new: &u128) {
+		TEN_TO_FOURTEEN.with(|v| {
+			let mut members = v.borrow_mut();
+			members.push(*new);
+			members.sort();
+		})
+	}
+}
+impl ContainsLengthBound for TenToFourteen {
+	fn max_len() -> usize {
+		TEN_TO_FOURTEEN.with(|v| v.borrow().len())
+	}
+	fn min_len() -> usize { 0 }
+}
+parameter_types! {
+	pub const ProposalBond: Permill = Permill::from_percent(5);
+	pub const ProposalBondMinimum: u64 = 1;
+	pub const SpendPeriod: u64 = 2;
+	pub const Burn: Permill = Permill::from_percent(50);
+	pub const DataDepositPerByte: u64 = 1;
+	pub const TreasuryModuleId: ModuleId = ModuleId(*b"py/trsry");
+	pub const MaximumReasonLength: u32 = 16384;
+}
+// impl pallet_treasury::Trait for Test {
+impl pallet_treasury::Trait<DefaultInstance> for Test {
```

Comment:
here you use an instance for pallet bounties in pallet treasury, this shouldn't be accepted by construct_runtime